docs/handbook/squeak.md

```diff
@@ -0,0 +1,165 @@
+
+So multi-homed systems (with multiple network interfaces) may have some strange behavior for TCP/IP networking inside Squeak.  This may require some experimentation to see what works.
+
+We've tested Squeak with a text-based installation with fixed IP address and hostname resolving is set up so that the fixed ip address corresponds to the hostname, and there were no other network interfaces on the test server.
+</div>
+
+
+<i class="fa fa-info-circle fa-lg" aria-hidden="true"></i>Installing Squeak
+<div class="well">
+</div>
+
+Squeak consists of a VM (virtual machine) and Smalltalk images.  You can find Smalltalk images at <a href= "http://files.squeak.org" target="_blank">files.squeak.org</a> or at <a href="http://cuis-smalltalk.org" target="_blank">cuis-smalltalk.org</a>.
+
+Because users of Smalltalk are normally saving their own custom image, there exist many other images on the internet to download.
+
+The images are cross-platform (in theory) and do not depend on OpenIndiana.  They can run on Windows or Linux or any other platform, in the same way as they will on OpenIndiana.
+
+The VM (virtual machine) is the interpreter that is OpenIndiana specific, and that runs (as some sort of interpreter or just-in-time compiler) the Smalltalk byte-code from the portable Smalltalk image.
+
+On OpenIndiana install as follows:
+
+# pkg list -af squeak-4
+NAME (PUBLISHER)                                  VERSION                    IFO
+
+runtime/squeak4 (userland)                        4.16.7                     ---
+
+# pkg install -v squeak-4
+
+During installation you may notice:
+
+Changed mediators:
+  mediator squeak:
+           version: None -> 4 (system default)
+
+With the squeak-4 package you can run Squeak 4.6 images.
+
+It is possible to simulteaously install squeak-5 as follows:
+
+# pkg mediator squeak
+MEDIATOR            VER. SRC. VERSION IMPL. SRC. IMPLEMENTATION
+squeak              system    4       system
+
+# pkg install -v squeak-5
+
+At the end of installation the hardlink squeak will point to the squeak-5 binary.  Note that the mediator changed:
+
+# pkg mediator squeak
+MEDIATOR            VER. SRC. VERSION IMPL. SRC. IMPLEMENTATION
+squeak              system    5       system     stack-spur
+
+The system-administrator can override the OpenIndiana choice by using the pkg set-mediator or pkg unset-mediator commands.
+
+Because both packages can be installed simulteaously it is possible to run side-by-side Squeak 4.6 and Squeak 5.3, Cuis Smalltalk or Squeak 6.0 alpha images.
+
+
+<i class="fa fa-info-circle fa-lg" aria-hidden="true"></i>Squeak Profile
+<div class="well">
+</div>
+
+When running squeak-5, you may notice that it is multi-threaded and is trying to set the priority of threads.  By default OpenIndiana will not allow the process to do that, but a special profile can be used to run squeak as follows:
+
+$ userattr profiles
+Squeak,System Administrator
```

Comment:
Should we fix this? Either patch out or deliver profile to allow squeak to run such actions as Basic User?
Better patch out.

Reply:
Please keep in mind that the Squeak project is looking at this from a completely different angle.   I'm just documenting one potential solution, which is to "pfexec squeak" run squeak under the Squeak profile.

Reply:
I think that Alexander is right. We shouldn't try to document two things in one place.
A better approach to document this is to separate the squeak introduction and the rights/profile management. Thus, deliver two documents and link from the squeak introduction to the rights/profile management documents.
My 2nd comment on your attempt to introduce Squeak/Cuis users to OpenIndiana: You have created Spur VMs. These are several times slower than a Cog (jitted) VM. This will propably create the idea that OI is slow. I would have preferred a Cog variant in order to get a comparable speed to other OS. I doubt that many Squeak/Cuis user prefer the old Spur VM over a Cog VM.

Reply:
Well as I wrote before, please relax the requirements and consider this "blog style" page as a starting point in the documentation.  If there is interest some users may consider improving this documentation and provide real documentation.    Also note that there is a manpage.  The manpage ("man squeak") was written I think by Ian Piumarta who did the original Squeak for Unix versions back in 1996.   The manpage has not been updated for a long time because I believe Ian Piumarta (who worked with Alan Kay) is already a little bit old now, I think Alan Kay himself is over 80 so those guys are not going to respond to my requests to update the manpage.    We must do a "best effort" here and not try to provide "perfection".   The blog-style document is just some notes on how I configured an OpenIndiana server (how I installed the server) and how this works for me for running Squeak, without an attempt to provide general documentation.   Besides general documentation would likely not be contributed to OpenIndiana but rather to the wiki.squeak.org.   I am also trying to get the people who maintain the wiki.squeak.org to update their docs on the OpenIndiana port.